### PR TITLE
Implementation

### PR DESCRIPTION
**Pull request by OscarHgg**

### CODE REVIEW

File: lib/cubit/meal_cubit.dart [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import 'package:bloc/bloc.dart';
```
Comment (OscarHgg) [inline lib/cubit/meal_cubit.dart:1]: Decided to try using Cubit as I have been trying to become better with it.

File: lib/cubit/meal_logic.dart [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+class _MealLogicState extends State<MealLogic> {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(body: BlocBuilder<MealCubit, MealState>(
+      builder: (context, state) {
+        if (state is MealInitial) {
+          return const WelcomeScreen();
+        }
+        if (state is MealLoading) {
```
Comment (OscarHgg) [inline lib/cubit/meal_logic.dart:25]: The Loading part became tricky in the end, when I realised I had to make the app more responsive. Solution is not perfect, but I learned a lot.

File: lib/screens/meal_detail_screen.dart [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+                  Container(
+                    width: double.maxFinite,
+                    height: MediaQuery.of(context).size.width * 0.95,
+                    child: Row(
+                      mainAxisAlignment: MainAxisAlignment.center,
+                      children: [
+                        ImageContainer(
+                          imagePath: meal.imageUrl,
+                          width: MediaQuery.of(context).size.width * 0.85,
+                          height: MediaQuery.of(context).size.height * 0.8,
```
Comment (OscarHgg) [inline lib/screens/meal_detail_screen.dart:72]: Using MediaQuery is what I relied on heavily to make it responsive for different devices.

File: lib/widgets/app_text.dart [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import 'package:flutter/material.dart';
+import 'package:shimmer/shimmer.dart';
+
```
Comment (OscarHgg) [inline lib/widgets/app_text.dart:3]: Created reusable widget for the text in the app

File: lib/widgets/image_container.dart [hunk 5]
```diff
@@ -0,0 +1,46 @@
+import 'package:flutter/material.dart';
+import 'package:shimmer/shimmer.dart';
+
```
Comment (OscarHgg) [inline lib/widgets/image_container.dart:3]: And image container, with shimmer functionality for loading

File: pubspec.lock [hunk 6]
```diff
@@ -62,32 +76,74 @@ packages:
     description: flutter
     source: sdk
     version: "0.0.0"
+  flutter_bloc:
```
Comment (OscarHgg) [inline pubspec.lock:79]: The necessary packages

**Comment by fenske** [comment]
Hi @OscarHgg. Anton from DevSkills is here.

It seems like there was some confusion as from what I understand you've put up the submission for your coding challenge on this boilerplate repo instead of your [assignment repo](https://github.com/ae50de9/meal-roulette-app-mobile-pagination-level-2_2c24664-35j6c5) (For the hiring team to review your work the PR should be created there).

Let me know if you need any help.

**Pull request closed without merging**